api/src/main/java/ca/bc/gov/educ/api/distribution/process/MergeProcess.java

```diff
@@ -0,0 +1,165 @@
+import org.springframework.web.reactive.function.client.WebClient;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Data
+@Component
+@NoArgsConstructor
+public class MergeProcess implements DistributionProcess {
+	
+	private static Logger logger = LoggerFactory.getLogger(MergeProcess.class);
+	
+	@Autowired
+    private ProcessorData processorData;
+
+	@Autowired
+	private GradStudentService gradStudentService;
+
+	@Autowired
+	GradValidation validation;
+
+	@Autowired
+	WebClient webClient;
+
+	@Autowired
+	EducDistributionApiConstants educDistributionApiConstants;
+
+	@Autowired
+	AccessTokenService accessTokenService;
+
+	@Autowired
+	SchoolService schoolService;
+
+	@Autowired
+	ReportService reportService;
+	
+	@Override
+	public ProcessorData fire() {				
+		long startTime = System.currentTimeMillis();
+		logger.info("************* TIME START  ************ "+startTime);
+		DistributionResponse response = new DistributionResponse();
+		ExceptionMessage exception = new ExceptionMessage();
+		Map<String,DistributionPrintRequest> mapDist = processorData.getMapDistribution();
+		processorData.setMapDistribution(mapDist);
+		int counter=0;
+		for (Map.Entry<String, DistributionPrintRequest> entry : mapDist.entrySet()) {
+			counter++;
+			String mincode = entry.getKey();
+			SchoolTrax schoolDetails = schoolService.getSchoolDetails(mincode,processorData.getAccessToken(),exception);
+			logger.info("School Details Acquired {}",schoolDetails.getSchoolName());
+			ReportRequest packSlipReq = reportService.preparePackingSlipData(schoolDetails,processorData.getBatchId());
+			DistributionPrintRequest obj = entry.getValue();
+			if(obj.getTranscriptPrintRequest() != null) {
+				TranscriptPrintRequest transcriptPrintRequest = obj.getTranscriptPrintRequest();
+				List<StudentCredentialDistribution> scdList = transcriptPrintRequest.getTranscriptList();
+				List<InputStream> locations=new ArrayList<InputStream>();
+				setExtraDataForPackingSlip(packSlipReq,"YED4",obj.getTotal(),scdList.size(),1,"Transcript",transcriptPrintRequest.getBatchId());
+				try {
+					locations.add(reportService.getPackingSlip(packSlipReq,processorData.getAccessToken(),exception).getInputStream());
+					logger.info("Packing Slip Added");
+					for (StudentCredentialDistribution scd : scdList) {
+						InputStreamResource transcriptPdf = webClient.get().uri(String.format(educDistributionApiConstants.getTranscript(),scd.getStudentID(),scd.getCredentialTypeCode(),"COMPL")).headers(h -> h.setBearerAuth(processorData.getAccessToken())).retrieve().bodyToMono(InputStreamResource.class).block();
+						locations.add(transcriptPdf.getInputStream());
+					}
+					PDFMergerUtility objs = new PDFMergerUtility();
+					Path path = Paths.get("C:/Users/s.karekkattumanasree/Downloads/"+mincode+"/");
```

Comment:
This path can be /tmp/mincode/ for now when it runs on Openshift. We'll work on setting up the volume for that.

Reply:
will change it :)